From 8b6d4da2a185e7b07c432267dc9fb80075f9565c Mon Sep 17 00:00:00 2001
From: Jonatan Lindblad <jlindblad@mythtv.org>
Date: Sun, 19 Apr 2015 23:41:00 +0200
Subject: [PATCH] Video output: Fix OpenGL 1 output after b9971146

(cherry picked from commit 2b01ed3894be8374b6c143449fa563fbd945da5a)
Signed-off-by: Stuart Auchterlonie <stuarta@squashedfrog.net>

Refs #12421
---
 mythtv/libs/libmythtv/openglvideo.cpp |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythtv/openglvideo.cpp b/mythtv/libs/libmythtv/openglvideo.cpp
index cf59d5b..e16af76 100644
--- a/mythtv/libs/libmythtv/openglvideo.cpp
+++ b/mythtv/libs/libmythtv/openglvideo.cpp
@@ -469,7 +469,9 @@ bool OpenGLVideo::AddFilter(OpenGLFilterType filter)
             success = false;
     }
 
-    if (success && (filter != kGLFilterNone))
+    if (success &&
+        (((filter != kGLFilterNone) && (filter != kGLFilterResize)) ||
+         ((gl_features & kGLSL) && (filter == kGLFilterResize))))
     {
         program = AddFragmentProgram(filter);
         if (!program)
@@ -1040,7 +1042,8 @@ void OpenGLVideo::PrepareFrame(bool topfieldfirst, FrameScanType scan,
 
         // enable fragment program and set any environment variables
         GLuint program = 0;
-        if ((type != kGLFilterNone))
+        if (((type != kGLFilterNone) && (type != kGLFilterResize)) ||
+            ((gl_features & kGLSL) && (type == kGLFilterResize)))
         {
             GLuint prog_ref = 0;
 
-- 
1.7.10.2

